### PR TITLE
Fix some nits in Configure stuff

### PR DESCRIPTION
**Pull request by richsalz**
trivial nits.  could be backported.

### CODE REVIEW

**Comment by levitte** [comment]
Re backport, have you checked if this change applies cleanly to 1.1.1?

**Comment by richsalz** [comment]
It does not patch cleanly.  I don't know if it is worth a separate PR to backport it.

**Comment by levitte** [comment]
OK. I took away the 1.1.1 label.
Although, at least the fix in 15-android.conf should be back ported.

**Comment by richsalz** [comment]
> Although, at least the fix in 15-android.conf should be back ported.
I'll put it on my TODO list once my doc PR's are merged, unless someone else gets to it first.

**Comment by mspncp** [comment]
Your pr fixes all declaration scope warnings

    "my" variable @foo masks earlier declaration in same scope

but I still get tons of unitialized value warnings using f4d2856217b6def5facb237d9343bfbc492aa62d. Dont' you get them too?


```
$ ./config
Operating system: x86_64-whatever-linux2
Use of uninitialized value in split at ./Configure line 669.
Use of uninitialized value in split at ./Configure line 670.
Use of uninitialized value in split at ./Configure line 669.
Use of uninitialized value in split at ./Configure line 670.
Configuring OpenSSL version 3.0.0-dev for target linux-x86_64
Using os-specific seed configuration
Use of uninitialized value $config{"cross_compile_prefix"} in concatenation (.) or string at (eval 6) line 78.
Use of uninitialized value $target{"sys_id"} in string ne at ./Configure line 1398.
Use of uninitialized value in concatenation (.) or string at ./Configure line 1403.
Use of uninitialized value in concatenation (.) or string at ./Configure line 1406.
Use of uninitialized value in concatenation (.) or string at ./Configure line 1444.
Use of uninitialized value $config{"CROSS_COMPILE"} in string eq at ./Configure line 1530.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value within %variables in concatenation (.) or string at ./Configure line 1813.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Use of uninitialized value in concatenation (.) or string at ./Configure line 2435.
Creating configdata.pm
Running configdata.pm
Creating Makefile
Use of uninitialized value $disabled{"threads"} in string eq at ./Configure line 2558.

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL file first)         ***
***                                                                ***
**********************************************************************
```

This is my perl version:

```
This is perl 5, version 30, subversion 1 (v5.30.1) built for x86_64-linux
(with 30 registered patches, see perl -V for more detail)
```

**Comment by richsalz** [comment]
My perl is 5.26  So I guess close this.

**Comment by mspncp** [comment]
> My perl is 5.26 So I guess close this.

Why do you want to close this? Wouldn't it be better to fix the warnings? @levitte are you getting [those warnings](https://github.com/openssl/openssl/pull/10965#issuecomment-579984757) too?

File: Configure [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 use 5.10.0;
 use strict;
+use warnings;
 use Config;
```
Comment (t-j-h) [inline Configure]: I would not turn on warnings without back-checking a wide range of perl versions ...

**Comment by richsalz** [comment]
I am closing this because it's not complete and I'm not going to proceed further.  If someone else wants to do it, they can create a new PR.

I am not worried about back-porting strict-fixes to older perls as perl never gets "more lax" in going forward.

**Pull request closed without merging**